### PR TITLE
Tweak Parse and SILGen to support loading swiftinterface files

### PR DESCRIPTION
**Pull request by jrose-apple**
Handle accessors, initializers, and deinitializers without bodies, which will all appear in textual interface files.

### CODE REVIEW

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

File: lib/Parse/ParseDecl.cpp [hunk 1]
```diff
@@ -6326,13 +6338,22 @@ parseDeclDeinit(ParseDeclOptions Flags, DeclAttributes &Attributes) {
     }
   }
 
+  if (Tok.is(tok::identifier)) {
```
Comment (harlanhaskins) [inline lib/Parse/ParseDecl.cpp]: Was this intended for this patch?

File: lib/SILGen/SILGen.cpp [hunk 2]
```diff
@@ -778,7 +778,7 @@ void SILGenModule::emitConstructor(ConstructorDecl *decl) {
 
     // If this constructor was imported, we don't need the initializing
```
Comment (harlanhaskins) [inline lib/SILGen/SILGen.cpp]: Can you update this comment to account for the behavior change?

**Review by harlanhaskins — APPROVED** [review]

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

**Comment by jrose-apple** [comment]
Whoops, this breaks `deinit \n override func test()`.

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

**Comment by jrose-apple** [comment]
…still got it wrong. I should just run the full test suite locally before the next try.

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

File: test/Parse/init_deinit.swift [hunk 3]
```diff
@@ -17,8 +17,8 @@ struct FooStructConstructorC {
 
 struct FooStructDeinitializerA {
   deinit // expected-error {{expected '{' for deinitializer}}
-  deinit x // expected-error {{deinitializers cannot have a name}} {{10-12=}}
-  deinit x() // expected-error {{deinitializers cannot have a name}} {{10-11=}}
+  deinit x // expected-error {{deinitializers cannot have a name}} {{10-12=}}  expected-error {{expected '{' for deinitializer}}
```
Comment (harlanhaskins) [inline test/Parse/init_deinit.swift]: Small drive-by diagnostic improvement too!

**Review by harlanhaskins — APPROVED** [review]
